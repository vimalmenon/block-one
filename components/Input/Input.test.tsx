import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import {Input} from "./";

describe("Input Component", () => {
    test("Input: Snapshot testing", () => {
        const wrapper = shallow(<Input value="testing"/>);
        expect(toJson(wrapper)).toMatchSnapshot();
    });
    test("Input: Snapshot testing with error", () => {
        const wrapper = shallow(<Input value="testing" error={true}/>);
        expect(toJson(wrapper)).toMatchSnapshot();
    });
    test("Input: Snapshot testing with No input value", () => {
        const wrapper = shallow(<Input error={true}/>);
        expect(toJson(wrapper)).toMatchSnapshot();
    });
    test("Input: Change Event Testing", () => {
        const onChange= jest.fn();
        const wrapper = shallow(<Input value="testing" onChange={onChange}/>);
        wrapper.find("input").simulate("change")
        expect(onChange).toHaveBeenCalled()
    });
});