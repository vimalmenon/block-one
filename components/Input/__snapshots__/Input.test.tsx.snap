// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input Component Input: Snapshot testing 1`] = `
<input
  className="py-2 px-4 block w-full sm:text-sm border rounded-md outline-none focus:border-blue-500 border-blue-100 focus:ring-blue-500"
  value="testing"
/>
`;

exports[`Input Component Input: Snapshot testing with No input value 1`] = `
<input
  className="py-2 px-4 block w-full sm:text-sm border rounded-md outline-none border-red-200 focus:border-red-500 focus:ring-red-500"
  value=""
/>
`;

exports[`Input Component Input: Snapshot testing with error 1`] = `
<input
  className="py-2 px-4 block w-full sm:text-sm border rounded-md outline-none border-red-200 focus:border-red-500 focus:ring-red-500"
  value="testing"
/>
`;
